tests/react/vanilla-utils/atomWithObservable.test.tsx

```diff
@@ -130,6 +130,54 @@ it('writable count state without initial value', async () => {
   await findByText('count: 3')
 })
 
+// See: https://github.com/pmndrs/jotai/discussions/2007
+it('writable count state without initial value and rxjs chain', async () => {
+  const single$ = new Subject<number>()
+  const double$ = single$.pipe(map((n) => n * 2))
+  const singleAtom = atomWithObservable(() => single$)
+  const doubleAtom = atomWithObservable(() => double$)
+
+  // THIS WORKS:
+  // const singleAndDoubleAtom = atom((get) =>
+  //   Promise.all([get(singleAtom), get(doubleAtom)])
+  // )
+
+  // in CounterValue:
+  // const [single, double] = useAtomValue(singleAndDoubleAtom)
+
+  const CounterValue = () => {
+    const single = useAtomValue(singleAtom)
+    const double = useAtomValue(doubleAtom)
```

Comment:
Thanks to @tmkx https://github.com/pmndrs/jotai/discussions/2007#discussioncomment-6890829, this doesn't work because the first hook suspends before reaching the second hook. It's basically how React Suspense works, and there are various workarounds one of which is `usePrepareAtoms`. https://github.com/pmndrs/jotai/discussions/2007#discussioncomment-6335514